packages/client/src/scenes/Match.tsx

```diff
@@ -85,19 +89,25 @@ export default class Match extends Component<IProps, IState> {
 
         // Connect
         try {
-            const host = window.document.location.host.replace(/:.*/, '');
-            const port = process.env.NODE_ENV !== 'production' ? Constants.WS_PORT : window.location.port;
-            const url = `${window.location.protocol.replace('http', 'ws')}//${host}${port ? `:${port}` : ''}`;
+            let definedRoomId = isNewRoom ? options.hathoraId : roomId
 
-            this.client = new Client(url);
+            this.connectionInfo = await pollConnectionInfo(definedRoomId)
+            
+            const url = 'wss://'+this.connectionInfo.host + ':' + this.connectionInfo.port
```

Comment:
can we use string interpolation with backticks? similar to ``wss://${host}:${port}`